File: ios/HybridSccKvPlatformContext.swift
import Foundation
import NitroModules

class HybridSccKvPlatformContext: HybridSccKvPlatformContextSpec {
  func getBaseDirectory() throws -> String {
    let urls = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask)
    guard let support = urls.first else {
      throw RuntimeError.error(withMessage: "Application Support directory not found")
    }
    let base = support.appendingPathComponent("react-native-scc", isDirectory: true)
    try? FileManager.default.createDirectory(at: base, withIntermediateDirectories: true)
    return base.path
  }
}

File: android/src/main/java/com/margelo/nitro/scckv/HybridSccKvPlatformContext.kt
package com.margelo.nitro.scckv

import com.margelo.nitro.NitroModules
import java.io.File

class HybridSccKvPlatformContext : HybridSccKvPlatformContextSpec() {
  override fun getBaseDirectory(): String {
    val context = NitroModules.applicationContext
      ?: throw Error("react-native-scc: no application context")
    val dir = File(context.filesDir, "react-native-scc")
    dir.mkdirs()
    return dir.absolutePath
  }
}
